### ISSUE TITLE
confused about the difference between `Dense` and `TimeDistributed(Dense())` 

### ISSUE THREAD

**Issue by songjun54cm** [fchollet/keras#5740]
This is my situation:
The sample data 'X' I have is a 3D tensor (N, C, F) as N = sample number, C=channel number, and F = feature size.
You can see that each sample in my data has a matrix which is a stack of C feature vectors.
I need to classify these data, and I have the ground-truth label 'Y' which is a stack of one-hot label with shape (N, K), where N is the sample number, and K is the class number.

I tried to build a Sequential model (mainly referenced the mnist mpl model) using X and Y as the data (X_train, Y_train, X_test, Y_test are splitted from 'X' and 'Y' respectively) as follows.

In order to get a prediction for each sample, I use a Lambda layer to average pooling over those channels, after two fully connected layer.

```
model = Sequential()
model.add(Dense(64, input_shape=(100,)))
model.add(Activation('relu'))
model.add(Dropout(0.2))
model.add(Dense(10))
model.add(Activation('relu'))

model.add(Lambda(function=lambda x: K.mean(x, axis=0),
                   output_shape=lambda shape: (shape[0],)+shape[2:]))

model.add(Activation('softmax'))
model.compile(loss='categorical_crossentropy',
              optimizer=RMSprop(),
              metrics=['accuracy'])

history = model.fit(X_train, Y_train,
                    batch_size=batch_size, nb_epoch=nb_epoch,
                    verbose=1, validation_data=(X_test, Y_test))
score = model.evaluate(X_test, Y_test, verbose=0)
```

The shape of X_train and X_test is (sample number, channel number, feature size), the shape of Y_train and Y_test is (sample number, class number).

However, the result is not good, the accuracy equals to a random guess result. So I tried to change the tow Dense layers in the model to TimeDistributed(Dense()). However the result still equals to a random guess result.

I am not very sure about the difference between Dense and TimeDistributed(Dense), what will happen if I use Dense in the model and what about TimeDistributed(Dense)??? What is the different here???

And finally, am I building the model in a wrong way ? why does it not work??

**Comment by unrealwill** [comment]
Hello,

I don't understand what you are trying to do, but you probably mean `axis=1`.

**Comment by songjun54cm** [comment]
In my situation, I want to build a classification model. The input data are many samples. The feature of each sample is a matrix contains fixed number of vectors. The output should be a classification result.

@unrealwill Thank you for your attention.
I finally make the model worked by changing two places (using Dense not Timedistributed(Dense()):
1. axis=1
2.optimizer='sgd'
In my experiments both these two settings have strong effect on the results.

However I am still puzzled about the difference between Dense and TimeDistributed(Dense()).

What indeed happen if I change Dense to TimeDistributed(Dense()) in my situation??

**Comment by mayou36** [comment]
Basically, your data will be processed independently (as it is assumed to be temporal different). Also see in the docs: https://keras.io/layers/wrappers/
And no, you don't want that as long as you are not working with time as a component
